Assets/Scripts/Save System/DataSaveManager.cs

```diff
@@ -0,0 +1,77 @@
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
+namespace Cosmobot
+{
+    public class DataSaveManager : SingletonSystem<DataSaveManager>
+    {
+        private GameData game_data;
+
+        [SerializeField]
+        private const string default_save_file_name = "save_file";
+        private List<ISaveableData> saveable_objects;
```

Comment:
This is C# not Rust xD

Use [C# naming conventions](https://learn.microsoft.com/en-us/dotnet/csharp/fundamentals/coding-style/identifier-names)
- const `PascalCase`
- private `camelCase`
- f. prefixes